src/System.Linq.Expressions/src/System/Dynamic/ExpandoObject.cs

```diff
@@ -309,7 +309,7 @@ private sealed class KeyCollectionDebugView
 
             public KeyCollectionDebugView(ICollection<string> collection)
             {
-                Debug.Assert(collection != null);
+                ContractUtils.RequiresNotNull(collection, nameof(collection));
```

Comment:
Hmm.  I don't think this really adds a lot of value, but it also only affects debug-only code paths and doesn't result in a lot of extra IL, so it doesn't really hurt, either.